algolia/algoliasearch-sfcc-b2c
# feat: Add promotionalPrices attribute

# What is Changed

- Added `promotions` attribute for keeping promotion prices and ids.
- Passed customer's active promotions and callouts to frontend
- Compared and displayed prices (Promotional or sale price)
- Displayed callout messages for applied promotion

## Testing
- Add `promotions` to additional attributes (There should be some active promos/campaigns)
- Run ProductIndex Job
- Check frontend
- Also, there are many different test cases but we simply need to check if promotional prices and promotional UI components work correctly for the following cases combination in UI:
   - There is an active promotion
   - There is no active promotion
   - Different record levels

## Pros
- Covering prices dynamically
- No complex configuration
- There is no additional effort for frontend integration
- Flexibility between different options (pricebook, promotions etc)

## Cons
- Each update on promos needs indexing
- Indexing needs when a campaign expires etc.
- Indexing time increased between %20-35
- Facets, sorting rules do not consider discounted prices
